File: src/main/kotlin/me/tomasan7/jecnadesktop/ui/component/LessonSpotView.java
package me.tomasan7.jecnadesktop.ui.component;

import javafx.scene.layout.VBox;
import me.tomasan7.jecnadesktop.data.LessonSpot;

public class LessonSpotView extends VBox
{
	private final LessonSpot lessonSpot;

	public LessonSpotView (LessonSpot lessonSpot)
	{
		this.lessonSpot = lessonSpot;

		getStyleClass().add("lesson-spot-view");
		setSpacing(1);

		lessonSpot.forEach(lesson -> getChildren().add(new LessonView(lesson)));
	}

	public LessonSpot getLessonSpot ()
	{
		return lessonSpot;
	}
}

File: src/main/kotlin/me/tomasan7/jecnadesktop/ui/component/LessonSpotView.kt
package me.tomasan7.jecnadesktop.ui.component

import javafx.scene.layout.VBox
import me.tomasan7.jecnaapi.data.Lesson
import me.tomasan7.jecnaapi.data.LessonSpot
import java.util.function.Consumer

class LessonSpotView(val lessonSpot: LessonSpot) : VBox()
{
    init
    {
        styleClass.add("lesson-spot-view")
        spacing = 1.0
        lessonSpot.forEach { children.add(LessonView(it)) }
    }
}